docs/customization/controller.rst

```diff
@@ -109,57 +110,42 @@ getting a list of recommended products from your external api.
 How to customize a Standard Controller:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Let's assume that you would like to send some kind of emails (which are not resources) after something has been purchased in your shop - to do this you should modify an ``afterPurchaseAction`` on the ``OrderController``.
+Let's assume that you would like to add some logic to the Homepage.
 
-1. Create a new Controller class under the ``AppBundle/Controller/Frontend`` namespace.
+1. Create a new Controller class under the ``AppBundle/Controller/Shop`` namespace.
 
-Run ``$ php bin/console debug:container sylius.controller.frontend.order``.
+Run ``$ php bin/console debug:container sylius.controller.shop.homepage``.
 
 Your class needs to extend this base class.
 
 .. code-block:: php
 
     <?php
 
-    namespace AppBundle\Controller\Frontend;
+    namespace AppBundle\Controller\Shop;
 
-    use Sylius\Bundle\WebBundle\Controller\Frontend\Account\OrderController as BaseOrderController;
-    use Sylius\Bundle\PayumBundle\Request\GetStatus;
     use Symfony\Component\HttpFoundation\Request;
     use Symfony\Component\HttpFoundation\Response;
+    use Sylius\Bundle\ShopBundle\Controller\HomepageController as BaseHomepageController;
 
-    class OrderController extends BaseOrderController
+    class HomepageController extends BaseHomepageController
     {
         /**
          * @param Request $request
          *
          * @return Response
          */
-        public function afterPurchaseAction(Request $request)
+        public function customAction(Request $request)
         {
-            $token = $this->getHttpRequestVerifier()->verify($request);
-            $this->getHttpRequestVerifier()->invalidate($token);
-
-            $status = new GetStatus($token);
-            $this->getPayum()->getGateway($token->getGatewayName())->execute($status);
-            $payment = $status->getFirstModel();
-            $order = $payment->getOrder();
-            $this->checkAccessToOrder($order);
-
-            $this->getOrderManager()->flush();
-
-            $emailManager = $this->get('sylius.email_manager.order');
-            $emailManager->sendConfirmationEmail($order);
-
-            return $this->redirectToRoute('sylius_checkout_thank_you');
+            // put your custom logic here
         }
     }
 
-2. The next thing you have to do is to override the ``sylius.controller.frontend.order.class`` parameter in ``app/config/services.yml``.
+2. The next thing you have to do is to override the ``sylius.controller.shop.homepage.class`` parameter in the ``app/config/services.yml``.
```

Comment:
We're not using this parameter anywhere, it won't work.

It is needed to override the whole `sylius.controller.shop.homepage` service definition.